assets/tutorials/multiple_feasible_solutions.jl

```diff
@@ -0,0 +1,240 @@
+#                                                                                #src
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     #src
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       #src
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    #src
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         #src
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  #src
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  #src
+# SOFTWARE.                                                                      #src
+
+# # Finding multiple feasible solutions 
+
+# **Author**: James Foster (@jd-foster)
+
+# This brief tutorial demonstrates how to formulate and solve a combinatorial problem
+# with multiple solutions. In fact, we will see how to find _all_ feasible solutions to our problem.
+# We will also see how to enforce an "all-different" constraint on a set of integer variables.
+
+# We start with some objects from recreational mathematics that could be called **"symmetric square sums"**.
+# Here are a few examples (inspired by [Number Squares](https://www.futilitycloset.com/2012/12/05/number-squares/)):
+"""
+    1529      2318      5219
+    5837      3790      2384
+    2340      1956      1867
+  + ____    + ____    + ____
+    9706      8064      9470
+"""
+
+# Notice how all the digits 0 to 9 are used at least once,
+# the first three rows sum to the last row,
+# and the columns in each are the same as the corresponding rows (forming a symmetric matrix).
+
+# We might ask: how many such squares are there?
+# In this tutorial, we will find out the answer
+# and see how to go about answering this and similar questions.
+
+# This tutorial uses the following packages:
+using JuMP
+import Gurobi
+
+# Here we are using [Gurobi](https://github.com/jump-dev/Gurobi.jl) 
+# since it provides the required functionality
+# for this example (i.e. finding multiple feasible solutions).
+# Gurobi is a commercial solver, that is, a paid license is needed
+# for those using the solver for commercial purposes. However there are
+# trial and/or free licenses available for academic and student users.
+
+# ## Model Specifics
+
+solver = Gurobi.Optimizer
+model = JuMP.Model(solver)
+
+# We need to set specific Gurobi parameters to enable the
+# [multiple solution functionality](https://www.gurobi.com/documentation/9.0/refman/finding_multiple_solutions.html).
+
+# The first setting turns on the exhaustive search mode for multiple solutions:
+JuMP.set_optimizer_attribute(model, "PoolSearchMode", 2)
+
+# The second sets a limit for the number of solutions found. 
+# Here the value 100 is an arbitrary but large enough whole number
+# for our particular model (but will depend on the application in general).
+JuMP.set_optimizer_attribute(model, "PoolSolutions", 100)
+
+## %%  #src
```

Comment:
Not sure what this is?

Reply:
This is just cell annotation is VS Code to help me step through.